core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java

```diff
@@ -9796,6 +9796,83 @@ private void checkLiteral2(String expression, String expected) {
         .withTrino().ok(expected);
   }
 
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-7048">[CALCITE-7048]
+   * The specified type conversion does not support complex types in Presto</a>. */
+  @Test void testPrestoFloatingPointNestedTypesCast() {
+
+    String query1 = "SELECT CAST(MAP[MAP[3.0,4.0],1.0]"
+        + " AS MAP<MAP<FLOAT, FLOAT>, FLOAT>)"
+        + " FROM \"employee\"";
+    String expectedPresto1 = "SELECT CAST(MAP (ARRAY[MAP (ARRAY[3.0], ARRAY[4.0])], ARRAY[1.0])"
```

Comment:
Why does the converted `Map` contains `Array`?

Reply:
This is to adapt to the Map type dialect of Presto. The Presto map needs to be converted to the form of Map(array(keys),array(values)). You can refer to the document https://prestodb.io/docs/current/language/types.html#map.
You can also refer to this test case: RelToSqlConverterTest##testMapValueConstructor

Reply:
I think `1.0` should not be an `Array`.

Reply:
You are right, please ignore this comment.